spec/octopus/serverspec/type/octopus_deploy_tentacle_spec.rb

```diff
@@ -0,0 +1,210 @@
+require 'spec_helper'
+require 'webmock/rspec'
+
+describe OctopusDeployTentacle do
+
+  let(:runner) { double ("runner") }
+
+  it "throws if `serverUrl` not supplied" do
+    expect { OctopusDeployTentacle.new(nil, "API-someapikey", "Tentacle1") }.
+        to raise_error(/serverUrl/)
+  end
+
+  it "throws if `apiKey` not supplied" do
+    expect { OctopusDeployTentacle.new("https://someserver.com", nil, "Tentacle1") }.
+        to raise_error(/apiKey/)
+  end
+
+  it "throws if the tentacle executable doesn't exist" do
+    allow(File).to receive(:exists?).and_return(false)
+    expect { OctopusDeployTentacle.new("https://someserver.com", "API-someapikey", "Tentacle1") }.
+        to raise_error(/tentacle\.exe/)
+  end
+
+  example_tentacle_response = get_api_example('/api/machines/all')
+
+  context "Server supports spaces" do
+
+    ex_supports_spaces = get_api_example('/api/2019.4.5')
+    before(:each) do
+      allow(File).to receive(:exists?).and_return(true)
+      stub_request(:get, "https://octopus.example.com/api/").
+          to_return(status: 200, body: ex_supports_spaces, headers: {})
+    end
+
+    # .registered_with_the_server()
+    it "returns false if the tentacle is not registered to the server" do
+      allow_any_instance_of(OctopusDeployTentacle).to receive(:`).and_return("D7E6B4CEEE0960CE944B92432605A2BAF14C7405")
```

Comment:
"receive(:`)"
☝️ this is why ruby is not my favourite... what on earth does that mean???

Reply:
Mocks up the backtick operator. I thought about putting that backtick call into a method and mocking that instead, because this does look a bit opaque...